src/etc/inc/ipsec.inc

```diff
@@ -2316,6 +2324,17 @@ function ipsec_setup_tunnels() {
 				$children[$cname]['local_ts'] = implode(",", $local_ts_spec);
 			}
 			if ($has_vti) {
+				// cleanup all vtimaps after IKEv1 or Split Connections
+				if (is_array($config['ipsec']['vtimap'])) {
+					$a_vtimap = &$config['ipsec']['vtimap'];
+					foreach ($a_vtimap as $id => $vtimap) {
+						if ($vtimap['reqid'] == $ph1ent['ikeid']) {
+							unset($a_vtimap[$id]);
```

Comment:
If this is making changes, should this not also set `$vtimap_save` to `true` so those changes get stored at the end?

Reply:
Now it's completely rewritten

No any extra write_config()
Correct creating/deleting vtimaps on IKEv1/IKEv2 switching
Correct creating/deleting vtimaps on Phase 2 mode changing